lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -1646,6 +1646,25 @@ class DecomposeAtenNewEmptyOp : public OpRewritePattern<AtenNewEmptyOp> {
 };
 } // namespace
 
+namespace {
+// Decompose `aten.pad` op into `aten.constant_pad_nd` op.
+class DecomposeAtenPadOp : public OpRewritePattern<AtenPadOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenPadOp op,
+                                PatternRewriter &rewriter) const override {
+
+    Value value = op.value();
+    if (!value)
```

Comment:
should check for NoneType here.

Also, need to guard for OptionalType. This is because we can't really handle optional type but only NoneType or FloatType.

Reply:
Added.